Libraries/LibWeb/HTML/FormControlInfrastructure.cpp

```diff
@@ -165,13 +165,13 @@ WebIDL::ExceptionOr<Optional<Vector<XHR::FormDataEntry>>> construct_entry_list(J
                 FileAPI::FilePropertyBag options {};
                 options.type = "application/octet-stream"_string;
                 auto file = TRY(FileAPI::File::create(realm, {}, String {}, options));
-                entry_list.append(XHR::FormDataEntry { .name = name.to_string(), .value = GC::make_root(file) });
+                entry_list.append(TRY(create_entry(realm, name.to_string(), Variant<GC::Ref<FileAPI::Blob>, String> { file }, {})));
             }
             // 2. Otherwise, for each file in selected files, create an entry with name and a File object representing the file, and append it to entry list.
             else {
                 for (size_t i = 0; i < file_element->files()->length(); i++) {
                     auto file = GC::Ref { *file_element->files()->item(i) };
-                    entry_list.append(XHR::FormDataEntry { .name = name.to_string(), .value = GC::make_root(file) });
+                    entry_list.append(TRY(create_entry(realm, name.to_string(), Variant<GC::Ref<FileAPI::Blob>, String> { file }, {})));
```

Comment:
Instead of repeating `Variant<..>` here, you should be able to just pass in `GC::Ref<FileAPI::Blob> { file }`.